src/components/Cards/EntityCard/variants/CompactEntityCard.tsx

```diff
@@ -91,7 +91,7 @@ export const CompactEntityCard: FC<Omit<EntityCardProps, 'type'>> = ({
           ))}
         </StyledParticipantsContainer>
         <StyledCompactEntityCardTitle>{title}</StyledCompactEntityCardTitle>
-        {rewardGroups && !!Object.keys(rewardGroups).length && (
+        {!!rewardGroups && !!Object.keys(rewardGroups).length && (
```

Comment:
To save on double negations, you could also use:

```suggestion
        {Object.keys(rewardGroups || {}).length > 0 && (
```